frappe/core/doctype/file/file.py

```diff
@@ -351,3 +351,67 @@ def check_file_permission(file_url):
 			return True
 
 	raise frappe.PermissionError
+
+def validate_space_limit(file_size):
+	"""Stop from writing file if max space limit is reached"""
+	from frappe.installer import update_site_config
+	from frappe.utils import cint
+	from frappe.utils.file_manager import MaxFileSizeReachedError
+
+	frappe_limits = get_limits()
+
+	if not frappe_limits:
+		return
+
+	if not frappe_limits.has_key('space_limit'):
+		return
+
+	# In Gigabytes
+	space_limit = flt(flt(frappe_limits['space_limit']) * 1024, 2)
+
+	# in Kilobytes
+	used_space = flt(frappe_limits['files_size']) + flt(frappe_limits['backup_size']) + flt(frappe_limits['database_size'])
+	file_size = file_size / (1024.0**2)
+
+	# Stop from attaching file
+	if flt(used_space + file_size, 2) > space_limit:
+		frappe.throw(_("You have exceeded the max space of {0} for your plan. {1} or {2}.").format(
+			"<b>{0}MB</b>".format(cint(space_limit)) if (space_limit < 1024) else "<b>{0}GB</b>".format(frappe_limits['space_limit']),
+			'<a href="#usage-info">{0}</a>'.format(_("Click here to check your usage")),
+			'<a href="#upgrade">{0}</a>'.format(_("upgrade to a higher plan")),
+		), MaxFileSizeReachedError)
+
+	# update files size in frappe subscription
+	new_files_size = flt(frappe_limits['files_size']) + file_size
+	set_limits({'files_size': file_size})
+
+def update_sizes():
+	from frappe.installer import update_site_config
+	# public files
+	files_path = frappe.get_site_path("public", "files")
+	files_size = flt(subprocess.check_output("du -ms {0}".format(files_path), shell=True).split()[0])
```

Comment:
Instead of `shell=True`, let's pass a list: `subprocess.check_output(['du', '-ms', files_path])`